core/src/main/java/com/graphhopper/routing/util/WheelchairFlagEncoder.java

```diff
@@ -0,0 +1,209 @@
+        allowedHighwayTags.add("cycleway");
+        allowedHighwayTags.add("unclassified");
+        allowedHighwayTags.add("road");
+        // disallowed in some countries
+        //allowedHighwayTags.add("bridleway");
+
+        maxPossibleSpeed = FERRY_SPEED;
+        speedDefault = MEAN_SPEED;
+        init();
+    }
+
+    @Override
+    public int getVersion() {
+        return 1;
+    }
+
+    /**
+     * Avoid some more ways than for pedestrian like hiking trails.
+     */
+    @Override
+    public EncodingManager.Access getAccess(ReaderWay way) {
+        if (way.hasTag("wheelchair", intendedValues)) {
+            return EncodingManager.Access.WAY;
+        }
+
+        if (way.getTag("sac_scale") != null) {
+            return EncodingManager.Access.CAN_SKIP;
+        }
+
+        if (way.hasTag("highway", avoidHighwayTags) && !way.hasTag("sidewalk", sidewalkValues)) {
+            return EncodingManager.Access.CAN_SKIP;
+        }
+
+        if (way.hasTag("surface", avoidSurfaces)) {
+            if (!way.hasTag("sidewalk", sidewalkValues)) {
+                return EncodingManager.Access.CAN_SKIP;
+            } else {
+                String sidewalk = way.getTag("sidewalk");
+                if (way.hasTag("sidewalk:" + sidewalk + ":surface", avoidSurfaces)) {
+                    return EncodingManager.Access.CAN_SKIP;
+                }
+            }
+        }
+
+        if (way.hasTag("smoothness", avoidSmoothness)) {
+            if (!way.hasTag("sidewalk", sidewalkValues)) {
+                return EncodingManager.Access.CAN_SKIP;
+            } else {
+                String sidewalk = way.getTag("sidewalk");
+                if (way.hasTag("sidewalk:" + sidewalk + ":smoothness", avoidSmoothness)) {
+                    return EncodingManager.Access.CAN_SKIP;
+                }
+            }
+        }
+
+        return super.getAccess(way);
+    }
+
+    @Override
+    public IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay way, EncodingManager.Access access, long relationFlags) {
+        if (access.canSkip()) {
+            return edgeFlags;
+        }
+
+        if (!access.isFerry()) {
+            String sacScale = way.getTag("sac_scale");
+            if (sacScale != null) {
+                accessEnc.setBool(false, edgeFlags, false);
+                accessEnc.setBool(true, edgeFlags, false);
+            } else {
+                speedEncoder.setDecimal(false, edgeFlags, MEAN_SPEED);
+                accessEnc.setBool(false, edgeFlags, true);
+                accessEnc.setBool(true, edgeFlags, true);
+            }
+        } else {
+            double ferrySpeed = getFerrySpeed(way);
+            setSpeed(false, edgeFlags, ferrySpeed);
+            accessEnc.setBool(false, edgeFlags, true);
+            accessEnc.setBool(true, edgeFlags, true);
+        }
+
+        return edgeFlags;
+    }
+
+    protected int handlePriority(ReaderWay way, int priorityFromRelation) {
```

Comment:
Can this method be removed? Looks identical...

Reply:
I changed the method to still use the super method, but also added wheelchair specific evaluation.